hal/halx86/generic/dma.c

```diff
@@ -1031,7 +1032,8 @@ HalpScatterGatherAdapterControl(IN PDEVICE_OBJECT DeviceObject,
 	ScatterGatherList = ExAllocatePoolWithTag(NonPagedPool,
 	                                          sizeof(SCATTER_GATHER_LIST) + sizeof(SCATTER_GATHER_ELEMENT) * ElementCount,
 											  TAG_DMA);
-	ASSERT(ScatterGatherList);
+	if (!ScatterGatherList)
+	    return DeallocateObject;
```

Comment:
Add a DPRINT1 here.

Reply:
Added, though no other such check in this file has a `DPRINT1()`.